src/osemgrep/cli_install_semgrep_pro/Install_semgrep_pro_subcommand.ml

```diff
@@ -75,24 +90,17 @@ let download_semgrep_pro (caps : < Cap.network ; .. >) platform_kind dest =
               m
                 "Visit https://semgrep.dev/products/pro-engine for more \
                  information.");
-          false
+          Lwt.return_false
       (* THINK: ??? is this raise for status? *)
-      | Error _ -> false
-      | Ok (body, { response; _ }) ->
-          (* Make sure no such binary exists. We have had weird situations
-           * when the downloaded binary was corrupted, and overwriting it did
-           * not fix it, but it was necessary to `rm -f` it.
-           *)
-          if Sys.file_exists dest then FileUtil.rm [ dest ];
-
-          (* TODO: does this matter if we don't have a progress bar? *)
-          let _file_size =
-            Cohttp.(Header.get (Response.headers response) "Content-Length")
-            |> Option.map int_of_string |> Option.value ~default:0
-          in
-
-          UFile.write_file (Fpath.v dest) body;
-          true)
+      | Ok { code; body = Error msg; _ } ->
+          Logs.err (fun m -> m "Error downloading Semgrep Pro: %d %s" code msg);
+          Lwt.return_false
+      | Error msg ->
+          Logs.err (fun m -> m "Error downloading Semgrep Pro: %s" msg);
+          Lwt.return_false)
+
+let download_semgrep_pro caps platform_kind dest =
+  Lwt_platform.run (download_semgrep_pro_async caps platform_kind dest)
```

Comment:
We're sure this is never called in an Lwt context? Seems like we might need to add the async version to the interface to me.

Reply:
Yea we definitely don't. This is a subcommand so isn't used by anything else.